### ISSUE TITLE
Image rotation in gallery does not work well

### ISSUE THREAD

**Issue by tikank** [getnikola/nikola#2418]
I have added an image to gallery. The image has orientation which is not 1 (normal), so gallery rotates it, but the result is not good. Size of the image is not changed and resulting image have black bands on sides and parts of the image is cropped.

If I follow code correctly, the rotation is done in nikola/image_processing.py and uses Image.rotate method. It should have expand=True flag to set the size correctly. Or maybe it would be better use Image.transpose method to rotate or flip the image.

**Comment by ralsina** [comment]
On what version of Nikola?

**Comment by tikank** [comment]
Nikola v7.7.11

**Comment by tikank** [comment]
I made a fix for this. I'll try to make pull request.

I also generated some test images using this script:

```
from PIL import Image, ImageDraw
import piexif

def save(orientation, im):
    im.save('{}_up.jpg'.format(orientation), format='jpeg')
    exif_data = { '0th' : { piexif.ImageIFD.Orientation : orientation } }
    im.save('{}_up_o.jpg'.format(orientation), format='jpeg', exif=piexif.dump(exif_data))

im = Image.new('RGBA', (100, 200), color=(255,255,255,255))
draw = ImageDraw.Draw(im)
draw.rectangle([(5,5), (95,195)], outline=(0,0,0,255))
draw.line([(50,80), (50,180)], fill=(0,0,0,255))
draw.line([(50,80), (20,110)], fill=(0,0,0,255))
draw.line([(50,80), (80,110)], fill=(0,0,0,255))
text = "This\nside\nup"
(w, h) = draw.textsize(text)
draw.text((50 - w/2, 20), text, fill=(0,0,0,255), align='center')

save(1, im.copy())
save(2, im.copy().transpose(Image.FLIP_LEFT_RIGHT))
save(3, im.copy().transpose(Image.ROTATE_180))
save(4, im.copy().transpose(Image.FLIP_LEFT_RIGHT).transpose(Image.ROTATE_180))
save(5, im.copy().transpose(Image.FLIP_LEFT_RIGHT).transpose(Image.ROTATE_90))
save(6, im.copy().transpose(Image.ROTATE_90))
save(7, im.copy().transpose(Image.FLIP_LEFT_RIGHT).transpose(Image.ROTATE_270))
save(8, im.copy().transpose(Image.ROTATE_270))

```

**Comment by ralsina** [comment]
Awesome!

**Comment by ralsina** [comment]
Fixed by #2421
